crates/task-impls/src/quorum_vote.rs

```diff
@@ -145,31 +143,20 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES> + 'static> HandleDepOutput
         )
         .await;
 
-        // TODO
-        // #[cfg(feature = "dependency-tasks")]
-        // {
-        //     let Some(proposal) = cur_proposal else {
-        //         error!("No proposal received, but it should be.");
-        //         return;
-        //     };
-        //     // For this vote task, we'll update the state in storage without voting in this function,
-        //     // then vote later.
-        //     update_state_and_vote_if_able::<TYPES, I>(
-        //         self.view_number,
-        //         proposal,
-        //         self.public_key.clone(),
-        //         self.consensus,
-        //         Arc::clone(&self.storage),
-        //         self.quorum_membership,
-        //         self.instance_state,
-        //         PhantomData,
-        //     )
-        //     .await;
-        // }
+        if !self.quorum_membership.has_stake(&self.public_key) {
+            debug!(
+                "We were not chosen for quorum committee on {:?}",
+                self.view_number
+            );
+            return;
+        }
 
         // Create and send the vote.
         let Some(leaf) = leaf else {
-            error!("Quorum proposal isn't validated, but it should be.");
+            debug!(
```

Comment:
This should probably be error level, since this would be an incorrect code path.